README.md

```diff
@@ -4,37 +4,680 @@ Sylvia is the old name meaning Spirit of The Wood.
+        assert_eq!(counter, contract::CounterResp { counter: 4});
+    }
+}
+```
+
+First of all, note the `contract` module I am using here - it is a slight change
+that doesn't match the previous code - I assume here that all the contract code
+sits in the `contract` module to make sure it is clear where the used type lies.
+So if I use `contract::something`, it is `something` in the module of the original
+contract (most probably sylvia-generated).
+
+First of all - we do not use `cw-multi-test` app directly. Instead we use the `sylvia`
+wrapper over it. It contains the original multi-test App internally, but it does
+it in an internally-mutable manner which makes it possible to avoid passing it
+everywhere around. It adds some overhead, but it should not matter for testing code.
+
+We are first using the `CodeId` type generated for every single Sylvia contract
+separately. Its purpose is to abstract storing the contract in the blockchain. It
+makes sure to create the contract object and pass it to the multitest.
+
+A contract's `CodeId` type has one particularly interesting function - the `instantiate`,
+which calls an instantiation function. It takes the same arguments as an instantiation
+function in the contract, except for the context that Sylvia's utilities would provide.
+
+The function doesn't instantiate contract immediately - instead, it returns what
+is called `InstantiationProxy`. We decided that we don't want to force users to set
+all the metadata - admin, label, and funds to send with every instantiation call,
+as in the vast majority of cases, they are irrelevant. Instead, the
+`InstantiationProxy` provides `with_label`, `with_funds`, and `with_amin` functions,
+which set those meta fields in the builder pattern style.
+
+When the instantiation is ready, we call the `call` function, passing the message
+sender - we could add another `with_sender` function, but we decided that as the
+sender has to be passed every single time, we can save some keystrokes on that.
+
+The thing is similar when it comes to execution messages. The biggest difference
+is that we don't call it on the `CodeId`, but on instantiated contracts instead.
+We also have fewer fields to set on that - the proxy for execution provides only
+the `with_funds` function.
+
+All the instantiation and execution functions return the
+`Result<cw_multi_test::AppResponse, ContractError>` type, where `ContractError`
+is an error type of the contract.
+
+## Interface items in multitest
+
+Because of implementation restrictions, calling methods from the contract interface
+looks slightly different:
+
+```rust
+#[test]
+fn member_test() {
+    let app = App::default();
+
+    let owner = "owner";
+    let member = "john";
+
+    let code_id = contract::multitest_utils::CodeId::store_code(&app);
+
+    let contract = code_id.instantiate(0)
+        .with_label("My contract")
+        .call(owner);
+
+    contract
+        .group()
```

Comment:
group_proxy()

Reply:
I really don't like this proxy suffix tbh :/ Just because we named type this way doesn't have to make the call have some suffixed added. Anyway - we will get rid of this call as soon as #109 is done.